cssom-view-1/Overview.bs

```diff
@@ -1070,21 +1070,18 @@ result of running these steps:
         <dt><a>caret range</a>
         <dd>null
     </dl>
-1. Otherwise, return a <a>caret position</a> where the
-    <a>caret range</a> is a collapsed
-    {{Range}} object for the position
-    where the text insertion point indicator would have been inserted when applying the <a>transforms</a> that apply to the descendants of the
-    <a>viewport</a>, and the other properties are set as follows:
-
-    <dl>
-        <dt><a>caret node</a>
-        <dd>The [=range/start node=]
-            of the <a>caret range</a>.
-
-        <dt><a>caret offset</a>
-        <dd>The [=range/start offset=] of
-            the <a>caret range</a>.
-    </dl>
+1. Otherwise:
+
+    1. Let <var>caretPosition</var> be a <a for=/>tuple</a> consisting of a <var>caretPositionNode</var> (a <a for=/>node</a>) and a <var>caretPositionOffset</var> (a non-negative integer) for the position where the text insertion point indicator would have been inserted when applying
+        the <a>transforms</a> that apply to the descendants of the <a>viewport</a>.
+    1. Let <var>startNode</var> be the <var>caretPositionNode</var> of the <var>caretPosition</var>, and let <var>startOffset</var> be the <var>caretPositionOffset</var> of the <var>caretPosition</var>.
+    1. While <var>startNode</var> is a [=node=], <var>startNode</var>'s [=tree/root=] is a [=shadow root=], and <var>startNode</var>'s [=tree/root=] is not a [=shadow-including inclusive ancestor=] of any of <var>shadowRoots</var>, repeat these steps:
+        1. Set <var>startOffset</var> to [=tree/index=] of <var>startNode</var>'s [=tree/root=]'s [=host=].
+        1. Set <var>startNode</var> to <var>startNode</var>'s [=tree/root=]'s [=host=]'s [=tree/parent=].</li>
+    1. Return a <a>caret position</a> with its properties set as follows:
+        1. <a>caret node</a> is set to <var>startNode</var>.
+        1. <a>caret offset</a> is set to <var>startOffset</var>.
+        1. <a>caret range</a> is set to a collapsed {{Range}} object which [=range/start node=] and [=range/end node=] are <var>startNode</var>, and which [=range/start offset=] and [=range/end offset=] are <var>startOffset</var>.
```

Comment:
Should this be removed now?

Reply:
Do you mean to remove the `caret range` from the algorithm? I think it's better to keep it in the spec even if we don't need to create the range right away. We should create the live range on demand, similar to https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/css/cssom/caret_position.cc;l=37

Reply:
Yes, okay to take care of this as a follow up issue, but can we add an issue into the current spec for this, so that implementors know that this may change.

Reply:
Filed https://github.com/w3c/csswg-drafts/issues/10230.

Reply:
Can we also add a section in the spec here that links to the issue? Like what we have here: https://www.w3.org/TR/css-highlight-api-1/#events

Reply:
Done.